src/plugins/controls/server/control_group/control_group_container_factory.ts

```diff
@@ -6,8 +6,10 @@
  * Side Public License, v 1.
  */
 
-import { EmbeddablePersistableStateService } from '../../../embeddable/common';
-import { EmbeddableRegistryDefinition } from '../../../embeddable/server';
+import type {
+  EmbeddablePersistableStateService,
+  EmbeddableRegistryDefinition,
+} from '../../../embeddable/common';
```

Comment:
Please let me clarify you some rules:
- If your file placed in `{your_plugin_id}/server/**/*.ts(x)` normally we should import from  `"{plugin}/server"`.
- If your file placed in `{your_plugin_id}/public/**/*.ts(x)` normally we should import from  `"{plugin}/public"`.
- If your file placed in `{your_plugin_id}/common/**/*.ts(x)` normally we should import from  `"{plugin}/common"`.

About this one. The following import `import { EmbeddablePersistableStateService } from '../../../embeddable/common';` was absolutely right and we don't needed to modify that. 

Please do re-export needed types from the common folder instead of removing it